client/src/main/java/com/paypal/selion/internal/platform/grid/LocalHub.java

```diff
@@ -42,9 +50,13 @@ static synchronized LocalServerComponent getSingleton() {
     synchronized LocalHub getLocalServerComponent() {
         if (instance == null) {
             instance = new LocalHub();
+            instance.setHost(new NetworkUtils().getNonLoopbackAddressOfThisMachine());
 
-            instance.setHost(new NetworkUtils().getIpOfLoopBackIp4());
-            instance.setPort(Integer.parseInt(Config.getConfigProperty(ConfigProperty.SELENIUM_PORT)));
+            int configHubPort = Integer.parseInt(Config.getConfigProperty(ConfigProperty.SELENIUM_PORT));
+            // Choose another free port when config port is not available.
+            int hubPort = LocalHub.pollPort(configHubPort) ? configHubPort : PortProber.findFreePort();
```

Comment:
I was able to fix my development environment and test this change. The `PortProber.pollPort(int)` returned true even when another grid was already running at port 4444. Hence add a private static method in `LocalHub`.